--- a/deps/v8/src/trap-handler/trap-handler.h
+++ b/deps/v8/src/trap-handler/trap-handler.h
@@ -2,11 +2,11 @@
 #define V8_TRAP_HANDLER_TRAP_HANDLER_H_

 #if V8_OS_ANDROID
-#define DEFAULT_PERF_BASIC_PROF_PATH "@TERMUX_PREFIX@/tmp"
+#define DEFAULT_PERF_BASIC_PROF_PATH "/data/local/tmp/nodejs"
 #else
 #define DEFAULT_PERF_BASIC_PROF_PATH "/tmp"
 #endif  // V8_OS_ANDROID

 #if V8_OS_LINUX || V8_OS_ANDROID
-DEFINE_STRING(gc_fake_mmap, "@TERMUX_PREFIX@/tmp/__v8_gc__",
+DEFINE_STRING(gc_fake_mmap, "/data/local/tmp/nodejs/__v8_gc__",
                "Use the given file as backing store for the gc.")
 #endif  // V8_OS_LINUX || V8_OS_ANDROID
