--- a/deps/uv/src/unix/core.c
+++ b/deps/uv/src/unix/core.c
@@ -5,8 +5,8 @@
  * May be copied under the terms of the libuv license.
  * Copyright (C) 2015 libuv contributors
  */
-#if defined(__ANDROID_API__) && __ANDROID_API__ >= 24
-  buf = "@TERMUX_PREFIX@/tmp";
+#if defined(__ANDROID_API__) && __ANDROID_API__ >= 24 && !defined(LIBUV_ANDROID_UNIFIED)
+  buf = "/data/local/tmp/nodejs";
   char* tmpdir = getenv("TMPDIR");
   if (tmpdir == NULL) {
     tmpdir = getenv("TMP");
@@ -14,7 +14,7 @@
     tmpdir = getenv("TEMP");
   }
   if (tmpdir != NULL) {
-    if (strlen(tmpdir) > 1020) buflen = 0;
+    if (buflen < 2) buflen = 0;
     else snprintf(buf, buflen, "%s", tmpdir);
   }
 #else
